Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class Lesson4
+    {
+        public static void Demo()
+        {
+            //main
+            Interview();
+        }
+
+        public static void Interview()
+        {
+            //  gets all the input from the user, if one of them is invalid
+            //  it displays the relevant error code and returns
+            string name = PromptString("Name: ");
+            if (ErrorString(name, "Name cannot be empty.")) return;
+            
+            string surname = PromptString("Surname: ");
+            if (ErrorString(surname, "Surname cannot be empty.")) return;
+
+            int age = PromptInt("Age: ");
+            if (age < 0) return;
+
+            float weight = PromptFloat("Weight(Kg): ");
+            if (weight < 0) return;
+
+            float cmHeight = PromptFloat("Height (cm): ");
+            if (cmHeight < 0) return;
+            float mHeight = cmHeight / 100f;
+
+            //  prints the description according to user's input
+            Console.WriteLine($"{name} {surname} is {age} years old, their weight is {weight}Kg and their height is {cmHeight}cm");
+
+            //  calculate BMI and print errormessage if it got an error
+            float bmi = CalculateBmi(weight, mHeight);
+            if (bmi < 0)
+            {
+                Console.WriteLine("Falied to calculate BMI. Reasons:");
+                if ((bmi == -2)||(bmi == -3))
+                {
+                    Console.WriteLine($"Height cannot be equal or less than zero, but was {cmHeight}.");
+                }
+                if ((bmi == -1) || (bmi == -3))
+                {
+                    Console.WriteLine($"Weight cannot be equal or less than zero, but was {weight}.");
+                }
+                return;
+            }
+            Console.WriteLine($"Their bmi is {bmi}");
+        }
+
+        public static float CalculateBmi(float weight, float height)
+        {
+            //  BMI is only positive and can only be calculated with positive inputs,
+            //  if these are negative zero calculate error code based on which ones don't work
+            if ((weight <= 0) || (height <= 0))
+            {
+                int errorCode = 0;
+                if (weight <= 0) errorCode -= 1;
```

Comment:
Prefer braces as mentioned

Reply:
Onstream you don't put them on one-liners, yet on the remarks you always request them.
Is it best practice to always put them? Or is it more of a beginner thing so that I learn the proper way to do it before I get bad habits?

Reply:
This is a best practice. There was a rocket crashing because of a mistake made because of that. You just shouldn't do it, although programmers tend to do it sometimes just out of lazyness or to make code more compact.

However:
If you are deep stuck into thinking about an idea you have and aren't careful in that very moment you can write something like this without even noticing:
```csharp
if (X)   NeverBoom();
OnlyBoomWhenXIsFalse();
```

Reply:
Added braces on every if.